### PR TITLE
Azure: disable caching as a temporary workaround to #6048

### PR DESCRIPTION
**Pull request by grlee77**
## Description

related to #6048

This PR is trying to temporarily comment out caching to see if the rest of the pipeline is still working.

**update**: instead of commenting these tasks out, there is apparently a `continueOnError` flag that we can set instead.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by grlee77** [comment]
This PR is not a fix, but shows that disabling caching is a plausible workaround until there is an upstream fix.

**Comment by grlee77** [comment]
I cancelled all of the non-Azure jobs here, so those failures are expected.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -63,6 +63,7 @@ jobs:
       key: '"skimage_data" | skimage/data/_registry.py'
       path: $(SKIMAGE_DATA_CACHE_FOLDER)
     displayName: Cache scikit-image data files
+    continueOnError: true
```
Comment (hmaarrfk) [inline azure-pipelines.yml:66]:
```suggestion
    # the line below (continueOnError: true) can be removed when the following issue is addressed:
    # https://github.com/scikit-image/scikit-image/issues/6048
    continueOnError: true
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -73,6 +74,7 @@ jobs:
         "Python$(PYTHON_VERSION)-$(ARCH)"
       path: $(PIP_CACHE_DIR)
     displayName: Cache pip packages
+    continueOnError: true
```
Comment (hmaarrfk) [inline azure-pipelines.yml:77]:
```suggestion
    # the line below (continueOnError: true) can be removed when the following issue is addressed:
    # https://github.com/scikit-image/scikit-image/issues/6048
    continueOnError: true
```

**Review by hmaarrfk — APPROVED** [review]

**Comment by hmaarrfk** [comment]
@grlee77 I simply elaborated in the config file itself. I think it helps read it better.

**Comment by grlee77** [comment]
Thanks @hmaarrfk, will merge now to have CI in a working state again. The pip caching did occasionally succeed the last time around, but it is still flaky.

**Pull request merged**